src/Symfony/Bundle/FrameworkBundle/Templating/TemplateResponse.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Templating;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+class TemplateResponse implements TemplateResponseInterface
+{
+    private $template;
+    private $parameters;
+    private $statusCode;
+    private $headers;
+
+    public function __construct($template, array $parameters = array(), $statusCode = Response::HTTP_OK, array $headers = array())
```

Comment:
What about cookies and cache information for example?

http://symfony.com/doc/current/components/http_foundation.html#setting-cookies

Internally cookies are added to the `ResponseHeaderBag`:
```php
    public function setCookie(Cookie $cookie)
    {
        $this->cookies[$cookie->getDomain()][$cookie->getPath()][$cookie->getName()] = $cookie;
    }
```

I think for this feature to be really viable, it should be possible to have caching and cookie control as well (hence I was aiming for a builder)